### DISCUSSION TITLE
Is this obsolete since next now has a built-in generateMetadata method?

### DISCUSSION THREAD

**Discussion by futpib** [garmeeh/next-seo#1189] [General]
I think this package should include a comparison to [built-in metadata support](https://beta.nextjs.org/docs/api-reference/metadata#generatemetadata) in Next.js and, if it covers most or all of the functionality of `next-seo`, should advise for or against using it instead and/or advise which solution should be preferred when.

**Comment by garmeeh** [comment]
Hey @futpib great question! This came up on a recent issue also so will cross post my response here 😄 

I had the pleasure of meeting with the Next.js team on the new meta data api and was able to give early feedback on it.

A small % of next-seo will be replaced by the meta data api eventually. The rest of next-seo will be useful for JSON-LD. 

At the moment the new meta data api is only available in the app directory which is not yet stable or production ready. So this library will be still used in pages.

I don't see the core of this library being removed anytime soon as many enterprise sites rely on next-seo and I can't see them migrating to app directory` fully anytime soon. Here are the latest numbers for the installs on versions:

<img width="768" alt="CleanShot 2023-03-30 at 08 31 48@2x" src="https://user-images.githubusercontent.com/13333582/228762582-8a702877-5f9f-458a-99b5-7af32bc8ad73.png">


However I do agree I need to update the README to advise on using the new built in method if using the app directory. I have added this to my TODO and will get it updated over the weekend.
